src/internal/operators/share.ts

```diff
@@ -118,9 +117,14 @@ export function share<T>(options?: ShareConfig<T>): OperatorFunction<T, T> {
     subject.subscribe(subscriber);
 
     if (!connection) {
-      connection = from(source).subscribe({
-        next: (value) => subject!.next(value),
-        error: (err) => {
+      // We need to create a subscriber here - rather than pass an observer and
+      // assign the returned subscription to connection - because it's possible
+      // for reentrant subscriptions to the shared observable to occur and in
+      // those situations we don't want connection to be already-assigned so
+      // that we don't create another connection to the source.
```

Comment:
> in those situations we don't want connection to be already-assigned so that we don't create another connection to the source.

dont you mean this?
"in those situations we ~~don't~~ want connection to be already-assigned so that we don't create another connection to the source."